net/prefix.go

```diff
@@ -22,6 +22,12 @@ func (p *Prefix) Dedup() *Prefix {
 	return pfxc.get(p)
 }
 
+// DedupWithIP gets a copy of Prefix from the cache and dedups the IP part
+func (p *Prefix) DedupWithIP() *Prefix {
+	p.addr = p.addr.Dedup()
```

Comment:
Is it intended that this function alters the original Prefix?

Reply:
Yes. That's fine.